sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1694,4 +1697,355 @@ private[spark] object QueryCompilationErrors {
+         """)
+  }
+
+  def loadDataNotSupportedForDatasourceTablesError(tableIdentWithDB: String): Throwable = {
+    new AnalysisException(s"LOAD DATA is not supported for datasource tables: $tableIdentWithDB")
+  }
+
+  def loadDataNoPartitionSpecProvidedError(tableIdentWithDB: String): Throwable = {
+    new AnalysisException(s"LOAD DATA target table $tableIdentWithDB is partitioned, " +
+      s"but no partition spec is provided")
+  }
+
+  def loadDataNumberColsNotMatchError(
+      tableIdentWithDB: String,
+      partitionSize: Int,
+      targetTableSize: Int): Throwable = {
+    new AnalysisException(s"LOAD DATA target table $tableIdentWithDB is partitioned, " +
+      s"but number of columns in provided partition spec ($partitionSize) " +
+      s"do not match number of partitioned columns in table " +
+      s"($targetTableSize)")
+  }
+
+  def loadDataButPartitionSpecWasProvidedError(tableIdentWithDB: String): Throwable = {
+    new AnalysisException(s"LOAD DATA target table $tableIdentWithDB is not " +
+      s"partitioned, but a partition spec was provided.")
+  }
+
+  def loadDataInputPathNotExistError(path: String): Throwable = {
+    new AnalysisException(s"LOAD DATA input path does not exist: $path")
+  }
+
+  def truncateTableOnExternalTablesError(tableIdentWithDB: String): Throwable = {
+    new AnalysisException(
+      s"Operation not allowed: TRUNCATE TABLE on external tables: $tableIdentWithDB")
+  }
+
+  def truncateTablePartitionNotSupportedOnTableNotPartitionedError(
+      tableIdentWithDB: String): Throwable = {
+    new AnalysisException(s"Operation not allowed: TRUNCATE TABLE ... PARTITION is not supported" +
+      s" for tables that are not partitioned: $tableIdentWithDB")
+  }
+
+  def failToTruncateTableWhenRemovingDataError(
+      tableIdentWithDB: String,
+      path: Path,
+      e: Throwable): Throwable = {
+    new AnalysisException(s"Failed to truncate table $tableIdentWithDB when " +
+        s"removing data of the path: $path because of ${e.toString}")
+  }
+
+  def descPartitionNotAllowedOnTempView(table: String): Throwable = {
+    new AnalysisException(s"DESC PARTITION is not allowed on a temporary view: $table")
+  }
+
+  def descPartitionNotAllowedOnView(table: String): Throwable = {
+    new AnalysisException(s"DESC PARTITION is not allowed on a view: $table")
+  }
+
+  def showPartitionNotAllowedOnTableNotPartitionedError(tableIdentWithDB: String): Throwable = {
+    new AnalysisException(
+      s"SHOW PARTITIONS is not allowed on a table that is not partitioned: $tableIdentWithDB")
+  }
+
+  def showCreateTableNotSupportedOnTempView(table: String): Throwable = {
+    new AnalysisException(s"SHOW CREATE TABLE is not supported on a temporary view: $table")
+  }
+
+  def showCreateTableFailToExecuteUnsupportedFeatureError(table: CatalogTable): Throwable = {
+    new AnalysisException("Failed to execute SHOW CREATE TABLE against table " +
```

Comment:
We could use
```
s"""
   |...
 """.stripMargin
```

Reply:
how can we use stripMargin.replace("\n", " ") here, since there are "\n" in this message?